Userland/Utilities/realsize.cpp

```diff
@@ -0,0 +1,114 @@
+#include <AK/Optional.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+
+static Optional<u64> use_integer_with_char_suffix(StringView argument, size_t suffix_length)
+{
+    if (suffix_length >= argument.length())
+        return {};
+    argument = argument.substring_view(0, argument.length() - suffix_length);
+
+    Optional<u64> numeric_optional = argument.to_number<u64>();
+    return numeric_optional;
+}
+
+static ErrorOr<size_t> handle_number_with_one_char_suffix(AK::HumanReadableBasedOn human_readable_based_on, StringView argument)
+{
+    VERIFY(human_readable_based_on == AK::HumanReadableBasedOn::Base2
+        || human_readable_based_on == AK::HumanReadableBasedOn::Base10);
+    auto numeric_value_without_suffix = use_integer_with_char_suffix(argument, 1).value();
+    u64 suffix_multiplier = 1;
+    auto suffix = argument[argument.length() - 1];
+
+    switch (suffix) {
+    case 'K':
+        suffix_multiplier = human_readable_based_on == AK::HumanReadableBasedOn::Base2 ? KiB : KB;
+        break;
+    case 'M':
+        suffix_multiplier = human_readable_based_on == AK::HumanReadableBasedOn::Base2 ? MiB : MB;
+        break;
+    case 'G':
+        suffix_multiplier = human_readable_based_on == AK::HumanReadableBasedOn::Base2 ? GiB : GB;
+        break;
+    case 'T':
+        suffix_multiplier = human_readable_based_on == AK::HumanReadableBasedOn::Base2 ? TiB : TB;
+        break;
+    default:
+        return Error::from_string_literal("Unknown size suffix");
+    }
+
+    if (Checked<u64>::multiplication_would_overflow(numeric_value_without_suffix, suffix_multiplier))
+        return Error::from_string_literal("Numeric value multiplication would overflow");
+
+    return numeric_value_without_suffix * suffix_multiplier;
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    StringView argument;
+
+    bool use_base10 = false;
+    bool use_base2 = false;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help(
+        "Show the 'real' size of a number with a suffix (possibly).");
+    args_parser.add_option(use_base10, "Use base 10 sizes (Kilobytes, Megabytes, Gigabytes)", "base10", 0);
+    args_parser.add_option(use_base2, "Use base 2 sizes (Kibibytes, Mebibytes, Gibibytes)", "base2", 0);
+    args_parser.add_positional_argument(argument, "Number with possibly a suffix", "number");
+    args_parser.parse(arguments);
+
+    if (argument.is_null() || argument.is_empty())
+        return Error::from_string_literal("Invalid value");
+
+    if (use_base2 && use_base10)
+        return Error::from_string_literal("Only one numerical base should be used");
+
+    if (use_integer_with_char_suffix(argument, 1).has_value()) {
+        if (!(use_base2 || use_base10))
+            return Error::from_string_literal("A numerical base should be used");
```

Comment:
This message feels a little bit vague for people who aren't familiar with the utility. I would suggest something a bit more explicit like: `Either --base2 or --base10 must be used with a single character suffix`